src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -1555,12 +1589,55 @@ static void window_tile_inspector_invalidate(rct_window* w)
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED].bottom = GBBB(propertiesAnchor, 2) - 3;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_INCREASE].top = GBBT(propertiesAnchor, 2) + 4;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_INCREASE].bottom = GBBB(propertiesAnchor, 2) - 4;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_DECREASE].top = GBBT(propertiesAnchor, 2) + 4;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_DECREASE].bottom = GBBB(propertiesAnchor, 2) - 4;
+                    w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].top = GBBT(propertiesAnchor, 3);
+                    w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].bottom = GBBB(propertiesAnchor, 3);
+                    w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].top = GBBT(propertiesAnchor, 4);
+                    w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].bottom = GBBB(propertiesAnchor, 4);
+                    w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].top = GBBT(propertiesAnchor, 5);
+                    w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].bottom = GBBB(propertiesAnchor, 5);
+                }
+                else
+                {
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED].type = WWT_EMPTY;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_INCREASE].type = WWT_EMPTY;
+                    w->widgets[WIDX_TRACK_SPINNER_SPEED_DECREASE].type = WWT_EMPTY;
+                    w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].top = GBBT(propertiesAnchor, 2);
+                    w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].bottom = GBBB(propertiesAnchor, 2);
+                    w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].top = GBBT(propertiesAnchor, 3);
+                    w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].bottom = GBBB(propertiesAnchor, 3);
+                    w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].top = GBBT(propertiesAnchor, 4);
+                    w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].bottom = GBBB(propertiesAnchor, 4);
+                }
+            }
+            else
+            {
+                w->widgets[WIDX_TRACK_SPINNER_SPEED].type = WWT_EMPTY;
+                w->widgets[WIDX_TRACK_SPINNER_SPEED_INCREASE].type = WWT_EMPTY;
+                w->widgets[WIDX_TRACK_SPINNER_SPEED_DECREASE].type = WWT_EMPTY;
+                w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].top = GBBT(propertiesAnchor, 2);
+                w->widgets[WIDX_TRACK_CHECK_CHAIN_LIFT].bottom = GBBB(propertiesAnchor, 2);
+                w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].top = GBBT(propertiesAnchor, 3);
+                w->widgets[WIDX_TRACK_CHECK_BLOCK_BRAKE_CLOSED].bottom = GBBB(propertiesAnchor, 3);
+                w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].top = GBBT(propertiesAnchor, 4);
+                w->widgets[WIDX_TRACK_CHECK_IS_INDESTRUCTIBLE].bottom = GBBB(propertiesAnchor, 4);
+            }
```

Comment:
This entire if/(if/else)/else block is there because you're hiding the widget for track elements that aren't boosters or brakes, right? Imagine how this scales if we add another widget that needs hiding. The code would be much simpler if the widget simply gets disabled when it cannot be used for a specific element, just like how it's done with the frame animation for small walls for instance. It'd also prevent the interface from changing for elements of the same type.